tests/basic.test.tsx

```diff
@@ -344,3 +344,28 @@ it('render from outside', async () => {
   fireEvent.click(getByText('toggle'))
   await findByText('count: 1')
 })
+
+it('should not trigger re-render when mutating object wrapped in ref', async () => {
+  const obj = proxy({ nested: ref({ count: 0 }) })
+
+  const Counter: React.FC = () => {
+    const snapshot = useProxy(obj)
+    return (
+      <>
+        <div>count: {snapshot.nested.count}</div>
+        <button onClick={() => ++obj.nested.count}>button</button>
+      </>
+    )
+  }
+
+  const { getByText, findByText } = render(
+    <StrictMode>
+      <Counter />
+    </StrictMode>
+  )
+
+  await findByText('count: 0')
+
+  fireEvent.click(getByText('button'))
+  await findByText('count: 0')
```

Comment:
I guess this check is weak. It doesn't guarantee that the count keeps `0`, does it?

Reply:
can you double check if this fails if you remove `ref`? I might be wrong though.

Reply:
yeah, it does still pass without ref 🤔 what would be your recommendation here to make the test stronger?

Reply:
added the wait to make the assertion of findByText, wait for the state update to be flushed. now it fails with no ref :)

Reply:
Yeah, I wasn't so sure about the best way.
Promise.resolve() is a nice hack, even though it depends on some implementation details. We are doing it in other tests, anyway.
Good to go. And hope someone has a better idea in the future.

Reply:
Yeah, thanks! i'll see if theres a better solution and update this and the other tests! 👍